src/target/adiv5_jtag.c

```diff
@@ -57,7 +57,8 @@ void adiv5_jtag_dp_handler(const uint8_t dev_index)
 	bmda_jtag_dp_init(dp);
 #endif
 
-	if (jtag_devs[dev_index].jd_idcode == JTAG_IDCODE_ARM_DPv0)
+	if ((jtag_devs[dev_index].jd_idcode & (JTAG_IDCODE_PARTNO_MASK | JTAG_IDCODE_DESIGNER_MASK)) ==
```

Comment:
Given that we have a routine here specific to jtag adiv5, and it has access to the dp struct, since it's created here, we could do the idcode parsing here, that way we could avoid the duplicate macro we have here for DPv0, and would not need to pass idcode to adiv5_dp_init at all anymore.

I'm not sure that would make the code cleaner or more obfuscated, so I'm leaving this not as a suggestion but as a note.

Reply:
That's a fair point. Perhaps that's an enhancement we perform after this PR is merged and battle tested a bit.

Reply:
As this PR has been tested by various people hitting one of the couple of different bugs it fixes, we're more confident that the code here doesn't create new regressions so we'll work on applying this refactor as suggested.